katsdpcontroller/product_config.py

```diff
@@ -296,6 +308,34 @@ class CbfStream:
     instrument_dev_name: str
 
 
+class SimDigRawAntennaVoltageStream(Stream):
+    stream_type: ClassVar[str] = 'sim.dig.raw_antenna_voltage'
+
+    def __init__(self, name: str, src_streams: Sequence[Stream], *,
+                 adc_sample_rate: float,
+                 centre_frequency: float,
+                 band: str,
+                 antenna: katpoint.Antenna) -> None:
+        super().__init__(name, [])
+        self.adc_sample_rate = adc_sample_rate
+        self.centre_frequency = centre_frequency
+        self.band = band
+        self.antenna = antenna
+
+    @classmethod
+    def from_config(cls,
+                    options: Options,
+                    name: str,
+                    config: Mapping[str, Any],
+                    src_streams: Sequence['Stream'],
+                    sensors: Mapping[str, Any]) -> 'SimDigRawAntennaVoltageStream':
```

Comment:
`sensors` argument doesn't appear to be used?

Reply:
Correct. All the stream classes define this function, and there is generic code that looks up the class to build at run-time and calls its `from_config` method. So they all need to have the same signature.